HW7/phone_book.py

```diff
@@ -1,24 +1,92 @@
-# Створити телефонну книгу, яка матиме наступні команди:
-# stats: кількість записів
-# add: додати запис
-# delete <name>: видалити запис за іменем (ключем)
-# list: список всіх імен в книзі
-# show <name>: детальна інформація по імені
-
-phone_book = []
-
-
+import json
+import os
+import time
+from json import JSONDecodeError
+
+
+# Initial all log files (phone book, log file, log error). Checkup log files.
+def initialized_json(file_name):
+    if not os.path.isfile(file_name):
+        with open(file_name, "w") as file:
+            file.write('[]')
+
+    with open(file_name, "r") as file:
+        try:
+            json_data = file.read()
+            lst = json.loads(json_data)
+            return lst
+        except JSONDecodeError:
+            lst = []
+            return lst
+
+
+phone_book = initialized_json("phone_book.json")
+log_error = initialized_json("log_error.json")
+log_file = initialized_json("log_file.json")
+# print(phone_book)
+# print(log_error)
+# print(log_file)
+
+
+# Decorator which loging name of func and time when the func was called
+def name_time_decorator(func):
+    def wrapper(*args, **kwargs):
+        name_func = func.__name__
+        time_call = time.strftime('%X', time.localtime())
+
+        new_data = {
+                        "name": name_func,
+                        "time": time_call,
+                   }
+        rewrite_json("log_file.json", new_data)
+        return func(*args, **kwargs)
+    return wrapper
+
+
+# Func which rewrite all changes in program.
+def rewrite_json(file_name, new_data, delete=False):
+    with open(file_name, "r") as file:
+        try:
+            json_data = file.read()
+            data = json.loads(json_data)
+        except JSONDecodeError:
+            data = []
+
+    data.append(new_data)
+
+    new_json_data = json.dumps(data)
+    with open(file_name, "w") as file:
+        file.write(new_json_data)
+
+    if delete:
```

Comment:
I think it would be better to create a function specially for phone book change because flag `delete` is not used in other logs